packages/mdc-select/icon/foundation.ts

```diff
@@ -22,66 +22,59 @@
  */
 
 import {MDCFoundation} from '@material/base/foundation';
-import MDCSelectIconAdapter from './adapter';
+import {EventType} from '@material/base/index';
+import {MDCSelectIconAdapter} from './adapter';
 import {strings} from './constants';
 
+const CLICK_KEYDOWN_EVENTS: EventType[] = ['click', 'keydown'];
 
-/**
- * @extends {MDCFoundation<!MDCSelectIconAdapter>}
- * @final
- */
-class MDCSelectIconFoundation extends MDCFoundation {
-  /** @return enum {string} */
+class MDCSelectIconFoundation extends MDCFoundation<MDCSelectIconAdapter> {
   static get strings() {
     return strings;
   }
 
   /**
-   * {@see MDCSelectIconAdapter} for typing information on parameters and return
-   * types.
-   * @return {!MDCSelectIconAdapter}
+   * See {@link MDCSelectIconAdapter} for typing information on parameters and return types.
    */
-  static get defaultAdapter() {
-    return /** @type {!MDCSelectIconAdapter} */ ({
-      getAttr: () => {},
-      setAttr: () => {},
-      removeAttr: () => {},
-      setContent: () => {},
-      registerInteractionHandler: () => {},
-      deregisterInteractionHandler: () => {},
-      notifyIconAction: () => {},
-    });
+  static get defaultAdapter(): MDCSelectIconAdapter {
+    // tslint:disable:object-literal-sort-keys
+    return {
+      getAttr: () => null,
+      setAttr: () => undefined,
+      removeAttr: () => undefined,
+      setContent: () => undefined,
+      registerInteractionHandler: () => undefined,
+      deregisterInteractionHandler: () => undefined,
+      notifyIconAction: () => undefined,
+    };
+    // tslint:enable:object-literal-sort-keys
   }
 
-  /**
-   * @param {!MDCSelectIconAdapter} adapter
-   */
-  constructor(adapter) {
-    super(Object.assign(MDCSelectIconFoundation.defaultAdapter, adapter));
+  private savedTabIndex_!: string | null; // assigned in initialSyncWithDOM()
+  private readonly interactionHandler_!: EventListener; // assigned in initialSyncWithDOM()
 
-    /** @private {string?} */
-    this.savedTabIndex_ = null;
+  constructor(adapter?: Partial<MDCSelectIconAdapter>) {
+    super({...MDCSelectIconFoundation.defaultAdapter, ...adapter});
 
-    /** @private {function(!Event): undefined} */
+    this.savedTabIndex_ = null;
```

Comment:
Correct.

The compilation error is only an issue for classes that extend `MDCComponent`, because the `MDCComponent` constructor calls instance methods (`initialize()` and `initialSyncWithDOM()`) _before_ the subclass constructor runs, so any values set by those instance methods get overwritten with `null` in the subclass constructor 🤦‍♂️.

`MDCFoundation` doesn't do that, thankfully.

Reply:
Done